src/compiler/checker.ts

```diff
@@ -7834,7 +7835,9 @@ namespace ts {
                     members.set(leftProp.escapedName, getNonReadonlySymbol(leftProp));
                 }
             }
-            return createAnonymousType(undefined, members, emptyArray, emptyArray, stringIndexInfo, numberIndexInfo);
+            const result = createAnonymousType(undefined, members, emptyArray, emptyArray, stringIndexInfo, numberIndexInfo);
+            result.flags |= (left.flags | right.flags) & TypeFlags.PropagatingFlags;

```

Comment:
is this required as part of this change or is it drive-by fix up, a smarter way to propagate propagating flags?

Reply:
It's both cleanup and a necessary change because we only want to set the propagated flags when we are actually creating the type (as opposed to sharing an existing one).